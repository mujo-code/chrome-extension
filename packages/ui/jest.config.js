const base = require('../../config/jest.base.js')

module.exports = {
  ...base,
}
