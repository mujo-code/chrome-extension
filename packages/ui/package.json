{
  "name": "@mujo/ui",
  "version": "2.3.0",
  "main": "dist/index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/mujo-code/plugins.git"
  },
  "keywords": [
    "react"
  ],
  "files": [
    "dist/**/*"
  ],
  "author": "Jacob Lowe",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mujo-code/plugins/issues"
  },
  "homepage": "https://github.com/mujo-code/plugins#readme",
  "scripts": {
    "test": "npm run lint && jest",
    "fmt": "prettier '{src,public}/**/*.js' --write",
    "lint": "eslint '{src,public}/**/*.js'",
    "build": "babel src --out-dir dist",
    "prepublish": "npm run build",
    "only-test": "jest --collectCoverage"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@babel/preset-env": "^7.4.5",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.6.0",
    "@emotion/core": "^10.0.14",
    "@mujo/box": "^3.0.1",
    "@mujo/eslint-config": "^1.0.4",
    "@mujo/ingress": "^0.1.0",
    "@mujo/plugins": "^2.2.0",
    "@mujo/utils": "^2.3.0",
    "@testing-library/react": "^9.1.4",
    "@testing-library/react-hooks": "^3.1.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "eslint": "^5.16.0",
    "jest": "^24.9.0",
    "jest-environment-jsdom-fourteen": "1.0.1",
    "jest-image-snapshot": "^2.9.0",
    "jest-resolve": "24.9.0",
    "jest-watch-typeahead": "^0.3.1",
    "prettier": "^1.17.1",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-modal": "^3.10.1",
    "react-test-renderer": "^16.9.0"
  },
  "peerDependencies": {
    "@emotion/core": ">=10.0.14",
    "@mujo/box": "^3.0.1",
    "@mujo/ingress": ">=0.1.0",
    "@mujo/plugins": "*",
    "@mujo/utils": "*",
    "react": ">=16.8.6",
    "react-modal": ">=3.10.1"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "browserslist": [
    "last 3 Chrome versions"
  ],
  "dependencies": {
    "@mujo/jest-webextension-mock": "^3.5.0",
    "@tweenjs/tween.js": "^18.3.1",
    "babel-jest": "^24.9.0",
    "react-feather": "^2.0.3",
    "react-spring": "^8.0.27"
  },
  "gitHead": "1a24f20513b5e7fa7d2e3d0245142cc61a4ae337"
}
