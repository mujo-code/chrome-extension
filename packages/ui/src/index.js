export * from './components'
export * from './styles/colors'
export * from './hooks'
