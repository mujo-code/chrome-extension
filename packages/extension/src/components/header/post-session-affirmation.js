import { Box } from '@mujo/box'
import { HeaderL, useTheme } from '@mujo/ui'
import React, { useEffect, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import { TRANSLATION_FILE } from '../../constants'
import { HappyFace } from './happy-face'

const TIMEOUT = 5000

const randomN = n => Math.floor(Math.random() * n)

const getRandomAffirmation = (t, i) => {
  const messages = [
    t('youre-amazing'),
    t('youre-mindful'),
    t('youre-the-best'),
    t('youre-pro'),
  ]
  const number = typeof i === 'number' ? i : randomN(messages.length)
  return messages[number]
}

export const PostSessionAffirmation = ({ close, index }) => {
  const { t } = useTranslation(TRANSLATION_FILE)
  const { foregroundSecondary, highlight } = useTheme()
  const message = useMemo(() => getRandomAffirmation(t, index), [
    index,
    t,
  ])

  useEffect(() => {
    const timer = setTimeout(() => {
      close()
    }, TIMEOUT)
    return () => {
      clearTimeout(timer)
    }
  }, [close])

  return (
    <Box>
      <HappyFace
        foreground={highlight}
        background={foregroundSecondary}
        marginBottom="s"
      />
      <HeaderL>{message}</HeaderL>
    </Box>
  )
}
