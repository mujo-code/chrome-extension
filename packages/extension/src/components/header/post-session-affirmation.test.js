import { ColorThemeProvider } from '@mujo/ui'
import React from 'react'
import { create } from 'react-test-renderer'
import { PostSessionAffirmation } from './post-session-affirmation'

test('PostSessionAffirmation component matches snapshot', () => {
  const tree = create(
    <ColorThemeProvider>
      <PostSessionAffirmation />
    </ColorThemeProvider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
