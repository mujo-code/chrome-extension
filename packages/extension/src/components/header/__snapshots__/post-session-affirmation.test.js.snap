// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSessionAffirmation component matches snapshot 1`] = `
<div
  className="css-0"
>
  <svg
    className="css-smhgd9"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      className="css-0"
      fill="none"
      fillRule="nonzero"
      transform="translate(-1)"
    >
      <circle
        className="css-0"
        cx="59"
        cy="43"
        fill="white"
        r="43"
      />
      <path
        className="css-0"
        d="M109.405 37.11a4 4 0 1 1 7.05 3.782c-9.898 18.45-29.28 27.609-57.49 27.609s-47.593-9.159-57.49-27.61a4 4 0 1 1 7.05-3.781c8.34 15.55 24.936 23.39 50.44 23.39 25.504 0 42.099-7.84 50.44-23.39z"
        fill="saltBox"
      />
      <path
        className="css-0"
        d="M79 36.412c7.18 0 13.055 6.225 13.055-2.93 0-9.156-5.875-4-13.055-4s-12.937-5.294-12.937 3.89 5.757 3.04 12.937 3.04zM39 36.412c7.18 0 13.055 6.225 13.055-2.93 0-9.156-5.875-4-13.055-4s-12.937-5.294-12.937 3.89 5.757 3.04 12.937 3.04z"
        fill="saltBox"
      />
    </g>
  </svg>
  <h2
    className="css-1lzekp8"
  >
    You are a pro at this!
  </h2>
</div>
`;
